main/com.spotify/index.js

```diff
@@ -59,6 +61,14 @@ module.exports = class SpotifyDevice extends Tp.BaseDevice {
         this._state = new Map();
         this._queryResults = new Map();
         this._deviceState = new Map();
+
+        const code = `killall spotifyd; spotifyd --username ${this.state.profile.id} --device-name ${this.state.profile.id} --token ${this.accessToken}`;
+        let hasSpotifyd = true;
+        exec(code, (error, stdout, stderr) => {
+            if (error) hasSpotifyd = false;
+        });
+        if (hasSpotifyd)
+            this._deviceId = crypto.createHash('sha1').update(this.state.userId).digest('hex'); //same protocol spotifyd uses
```

Comment:
- You should not use `killall`: there is no guarantee that it is installed on all systems, and it is not safe. Instead, you should keep track of the object returned by [child_process.exec](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback) and use `.kill()` on that one instead.
- We should run this code only on the server platform (`this.platform.type === 'server'`).
- We should probably download the right version of spotifyd from github automatically (especially given that one needs your modified spotifyd), save it in a cache directory, and launch that version directly. That way one does not need to install it separately.
- You should never use `child_process.exec`, always use `child_process.execFile` or `child_process.spawn`. `exec` will pass the command line to a shell, which makes you vulnerable to command-line injection.
- You will probably want to pass `--no-daemon` to spotifyd, so it runs in foreground and we can kill it and restart it.